ui/lib/kv/addon/components/page/secrets/create.js

```diff
@@ -48,30 +49,61 @@ export default class KvSecretCreate extends Component {
     }
   }
 
+  validate() {
+    const dataValidations = this.args.secret.validate();
+    const metadataValidations = this.args.metadata.validate();
+    const state = { ...dataValidations.state, ...metadataValidations.state };
+    const failed = !dataValidations.isValid || !metadataValidations.isValid;
+    return { state, isValid: !failed };
+  }
+
   @task
   *save(event) {
     event.preventDefault();
-    try {
-      const { isValid, state, invalidFormMessage } = this.args.secret.validate();
-      this.modelValidations = isValid ? null : state;
-      this.invalidFormAlert = invalidFormMessage;
-      if (isValid) {
-        const { secret } = this.args;
-        yield this.args.secret.save();
-        this.flashMessages.success(`Successfully created secret ${secret.path}.`);
+    this.flashMessages.clearMessages();
+    this.errors = [];
+    const { isValid, state } = this.validate();
+    this.modelValidations = isValid ? null : state;
+    this.invalidFormAlert = isValid ? '' : 'There was a problem submitting this form.';
+
+    const { secret, metadata } = this.args;
+    // users must have kv/data create to create metadata in the UI
+    if (isValid) {
+      try {
+        // try saving secret data first
+        yield secret.save();
+        this.flashMessages.success(`Successfully saved secret data for: ${secret.path}.`);
+      } catch (error) {
+        this.errors.pushObject({ endpoint: 'kv/data', message: errorMessage(error) });
+      }
+
+      // only attempt to save metadata secret data was saved
+      if (secret.createdTime) {
+        try {
+          metadata.path = secret.path;
+          yield metadata.save();
+          this.flashMessages.success(`Successfully saved metadata.`);
+        } catch (error) {
```

Comment:
(Discussed via zoom - posting resolution here for posterity)

We do not because any metadata errors shouldn't prevent a transition and will only surface in a flash message